File: backend/app/schemas/return_.py
from datetime import datetime

from pydantic import BaseModel


class ReturnItemIn(BaseModel):
    product_id: int
    quantity: int


class ReturnCreate(BaseModel):
    returned_items: list[ReturnItemIn]
    new_items: list[ReturnItemIn] = []


class ReturnComplete(BaseModel):
    manager_pin: str


class ReturnItemOut(BaseModel):
    product_id: int
    quantity: int
    unit_price: float


class ReturnOut(BaseModel):
    id: int
    sale_id: int
    returned_items: list[ReturnItemOut]
    new_items: list[ReturnItemOut]
    net_amount: float
    status: str
    created_at: datetime
    completed_by: int | None = None
    completed_at: datetime | None = None

File: frontend/src/types/return.ts
// backend/app/schemas/return_.py ile birebir eşleşir.
export interface ReturnItemIn {
  product_id: number;
  quantity: number;
}

export interface ReturnItemOut {
  product_id: number;
  quantity: number;
  unit_price: number;
}

export interface ReturnOut {
  id: number;
  sale_id: number;
  returned_items: ReturnItemOut[];
  new_items: ReturnItemOut[];
  net_amount: number;
  status: string;
  created_at: string;
  completed_by: number | null;
  completed_at: string | null;
}
